api/kv_v2.go

```diff
@@ -671,3 +756,33 @@ func readThenWrite(ctx context.Context, client *Client, mountPath string, secret
 
 	return updatedSecret, nil
 }
+
+func toMetadataMap(metadata KVMetadataInput) (map[string]interface{}, error) {
+	md := make(map[string]interface{})
+
+	// The KVMetadataInput struct is designed to have pointer fields so that
+	// the user can easily express the difference between explicitly setting a
+	// field back to its zero value (e.g. false), as opposed to just having
+	// the field remain unchanged (e.g. nil). This way, they only need to pass
+	// the fields they want to change.
+	if metadata.MaxVersions != nil {
+		md[maxVersionsKey] = *(metadata.MaxVersions)
+	}
+	if metadata.CASRequired != nil {
+		md[casRequiredKey] = *(metadata.CASRequired)
+	}
+	if metadata.CustomMetadata != nil {
+		if len(metadata.CustomMetadata) == 0 { // empty non-nil map means delete all the keys
+			md[customMetadataKey] = nil
+		} else {
+			md[customMetadataKey] = metadata.CustomMetadata
+		}
+	}
+	if metadata.DeleteVersionAfter != nil {
+		derefVal := *metadata.DeleteVersionAfter
+		strDuration := derefVal.String()
+		md[deleteVersionAfterKey] = strDuration
```

Comment:
```suggestion
		md[deleteVersionAfterKey] = (*metadata.DeleteVersionAfter).String()
```
This might be easier to read